### PR TITLE
feat: Add Grafana Managed

### PR DESCRIPTION
**Pull request by rtrive**
### List of changes
Add grafana managed and cname record

### Motivation and context
Gilda k6 resources

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [x] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/core/grafana.tf [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+
+resource "azurerm_dashboard_grafana" "example" {
+  name                              = format("%s-grafana", local.project)
```
Comment (diegolagospagopa) [inline src/core/grafana.tf:4]: Please use the interpolation format

File: src/domains/load-test/01_grafana.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module "grafana_managed" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//grafana?ref=feature/grafana"
+
+  name = format("%s-grafana", var.env_short)
```
Comment (diegolagospagopa) [inline src/domains/load-test/01_grafana.tf:4]: please use interpolation

File: src/domains/load-test/01_grafana.tf [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  resource_group_name = var.terraform_remote_state_core.resource_group_name
```
Comment (diegolagospagopa) [inline src/domains/load-test/01_grafana.tf:6]: Please put the grafana in its own resource group, and not use the one used by terraform state

**Review by diegolagospagopa — CHANGES_REQUESTED** [review]

File: src/domains/load-test/01_grafana.tf (hunk 3 above)
Comment (rtrive) [reply src/domains/load-test/01_grafana.tf:6]: I created a new one. Is it right?

File: src/core/grafana.tf (hunk 1 above)
Comment (rtrive) [reply src/core/grafana.tf:4]: I copied from all the code that is inside the repo; example: https://github.com/pagopa/pagopa-infra/blob/4243840adcf9bc46444cfe27588c6e6d3900000e/src/core/apim.tf#L4.

Is interpolation still ok?

**Review by pasqualespica — APPROVED** [review]
great work 🚀 
I'd only mention in PR description use of `terraform-azurerm-v3.git`

File: src/domains/load-test/01_grafana.tf [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  resource_group_name = azurerm_resource_group.load_test.name
+
+  tags = var.tags
+
+}
+
+resource "azurerm_dns_cname_record" "grafana" {
+
+  name                = "grafana.${var.dns_zone_prefix}.${var.external_domain}"
```
Comment (diegolagospagopa) [inline src/domains/load-test/01_grafana.tf:21]: This is only the record dns, not the entire fqdn

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/dns_cname_record

Comment (diegolagospagopa) [reply src/domains/load-test/01_grafana.tf:21]:
```suggestion
name                = "grafana"
```

**Comment by diegolagospagopa** [comment]
Good job with the refactors, i have one question. This grafana will be used only for load-test?

if not. maybe i think is better to call the resource group/folders/state with another name.

File: src/domains/load-test/01_grafana.tf (hunk 3 above)
Comment (diegolagospagopa) [reply src/domains/load-test/01_grafana.tf]: yes

File: src/core/grafana.tf (hunk 1 above)
Comment (diegolagospagopa) [reply src/core/grafana.tf]: yes it is

**Comment by rtrive** [comment]
> Good job with the refactors, i have one question. This grafana will be used only for load-test?
> 
> if not. maybe i think is better to call the resource group/folders/state with another name.

Got your point, I used `load-test` because the scope is limited to the `gilda` and I don't know if Grafana will be useful for others

**Comment by diegolagospagopa** [comment]
> > Good job with the refactors, i have one question. This grafana will be used only for load-test?
> > if not. maybe i think is better to call the resource group/folders/state with another name.
> 
> Got your point, I used `load-test` because the scope is limited to the `gilda` and I don't know if Grafana will be useful for others

ok perfect

**Review by diegolagospagopa — APPROVED** [review]

**Comment by diegolagospagopa** [comment]
Great work, i think is ok and can be merged

**Pull request merged**